exporter/exporterhelper/internal/base_exporter.go

```diff
@@ -196,6 +196,17 @@ func WithQueue(cfg queuebatch.Config) Option {
 		if o.queueBatchSettings.Encoding == nil {
 			return errors.New("WithQueue option is not available for the new request exporters, use WithQueueBatch instead")
 		}
+		// Automatically configure partitioner if MetadataKeys is set
+		if len(cfg.MetadataKeys) > 0 {
+			if o.queueBatchSettings.Partitioner != nil {
+				return errors.New("cannot use metadata_keys when a custom partitioner is already configured")
+			}
+			if o.queueBatchSettings.MergeCtx != nil {
+				return errors.New("cannot use metadata_keys when a custom merge function is already configured")
+			}
```

Comment:
Do we really need to check this here and in xexporterhelper.WithQueueBatch? Could we check interpret the config in WithQueueBatch below, and check for the conflict there?

Reply:
Last update in https://github.com/open-telemetry/opentelemetry-collector/pull/14139#discussion_r2565312763